src/main/java/org/spongepowered/common/data/processor/data/WetDataProcessor.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.processor.data;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Optional;
+
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.DataHolder;
+import org.spongepowered.api.data.DataTransactionBuilder;
+import org.spongepowered.api.data.DataTransactionResult;
+import org.spongepowered.api.data.key.Key;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.ImmutableWetData;
+import org.spongepowered.api.data.manipulator.mutable.WetData;
+import org.spongepowered.api.data.merge.MergeFunction;
+import org.spongepowered.api.data.value.BaseValue;
+import org.spongepowered.api.entity.EntityType;
+import org.spongepowered.api.item.ItemTypes;
+import org.spongepowered.common.data.ImmutableDataCachingUtil;
+import org.spongepowered.common.data.manipulator.immutable.ImmutableSpongeWetData;
+import org.spongepowered.common.data.manipulator.mutable.SpongeWetData;
+import org.spongepowered.common.data.processor.common.AbstractSpongeDataProcessor;
+import org.spongepowered.common.data.util.DataUtil;
+
+import net.minecraft.entity.passive.EntityWolf;
+import net.minecraft.item.ItemStack;
+
+public class WetDataProcessor extends AbstractSpongeDataProcessor<WetData, ImmutableWetData> {
```

Comment:
Rereading this processor after rebasing the commits into one, there's a few things you need to understand when implementing a `DataProcessor`:
- The DataProcessor should NEVER delegate back to a `ValueProcessor` at all costs. If you're being offered data already, simply set it, don't re-delegate for a key -> processor lookup just because.
- There are a lot of abstractions that are done with `AbstractEntitySingleDataProcessor` and `AbstractItemSingleDataProcessor` such that you can narrow down a focus on the actual data manipulation versus having to surround all of it with boilerplate code of `instanced` checks.
- You forgot to notice that `EntityWolf.isWet()` is just an inherited method from `Entity`, which isn't quite the same as `EntityWolf.isWet` or `EntityWolf.isShaking`.

All of these I'm fixing while merging since most of the rest of the work is done, but i'm leaving this note for guidance and future reference.